version: '3.8'

services:
  nginx:
    image: harbor.pyoseyeol.co.kr/docker_hub/library/nginx:1.22.0-alpine
    #image: harbor.pyoseyeol.co.kr/docker_hub/nginxinc/nginx-unprivileged:1.22.0-alpine
    container_name: nginx-reverse-proxy
    network_mode: host
    restart: always
    logging:
      driver: 'json-file'
      options:
        max-size: '100m'
        max-file: '5'
    healthcheck:
      test: >
        wget -q --tries=1 -O- http://localhost:80/health | grep -q 'true'
      interval: 15s
      timeout: 5s
      retries: 3
      start_period: 30s
    environment:
      TZ: 'Asia/Seoul'
    volumes:
      # default config
      - ./conf/nginx.conf:/etc/nginx/nginx.conf:ro
      - ./conf/conf.d/default.conf:/etc/nginx/conf.d/default.conf:ro
      # service dependent config
      - ./conf/conf.d/${CONF}.conf:/etc/nginx/conf.d/${CONF}.conf:ro
      - ./conf/cert/${CONF}:/etc/nginx/tls:ro
      # init shell
      - ./init_script/init.sh:/docker-entrypoint.d/init.sh:ro
      # log
      - ./logs:/var/log/nginx/logs